### ISSUE TITLE
Integrate a bot to enforce layout style guidelines

### ISSUE THREAD

**Issue by BenHenning** [oppia/oppia-android#994]
The guidelines mentioned https://github.com/oppia/oppia-android/wiki/Coding-style-guide#layout-files should be straightforward to enforce with a GitHub check via a custom action or bot. We might be able to leverage Oppia bot for this (once and if we integrate it with this repository), or look into a custom solution for it.

/cc @vinitamurthi since this falls under dev workflow

**Comment by seanlip** [comment]
@BenHenning We could just have a lint check script for this, right?

**Comment by BenHenning** [comment]
> [@BenHenning](https://github.com/BenHenning) We could just have a lint check script for this, right?

@seanlip yep but it would need to be a custom one. Some of the checks could be enforced with our static regex checker, but to actually make sure the correct convention is used in the correct place will likely require a custom script.

I suspect it's probably worth putting this on hold actually and repurposing it to enforce Jetpack Compose best practices (which are still being figured out). Compose will significantly change these rules so it probably doesn't make sense to build something now for layouts that will eventually go away (even on the order of years--we haven't run into many problems without this check in place, anyway).

I've updated https://github.com/oppia/oppia-android/issues/5422#issuecomment-2347068869 to include a line item for this exact purpose as that issue will probably be our long-term tracking for Compose. With that now tracked there and no further desire to implement non-Compose checks I think we can close this.
